### PR TITLE
MaKe Date Enhancement

### PR DESCRIPTION
**Pull request by stevenndungu**

### CODE REVIEW

**Comment by stevenndungu** [comment]
@danny could you have a look as I add something here to it..

**Comment by dannyparsons** [comment]
Sure. Remember I'm @dannyparsons not @ danny :)

**Comment by stevenndungu** [comment]
This is ready now

**Comment by dannyparsons** [comment]
@africanmathsinitiative/developers can someone review this?

**Comment by stevenndungu** [comment]
Fixes #2475

**Comment by Lunalo** [comment]
am reviewing it

File: instat/dlgMakeDate.vb [hunk 1]
```diff
@@ -23,7 +23,7 @@ Public Class dlgMakeDate
     Dim bUseSelectedColumn As Boolean = False
     Dim strSelectedColumn As String = ""
     Dim strSelectedDataFrame As String = ""
```
Comment (Lunalo) [inline instat/dlgMakeDate.vb]: all these should be public

Comment (stevenndungu) [inline instat/dlgMakeDate.vb]: Changed.

**Comment by stevenndungu** [comment]
@Lunalo I have made the changes requested.Can you have a check once more

**Comment by Lunalo** [comment]
@dannyparsons 

This is ready for review

**Comment by lilyclements** [comment]
I'm going to make minor edits to this first!

**Comment by lilyclements** [comment]
Made changes waiting to be merged.
Two bugs:

1. I clicked "Help..." and first time it crashed. When I tried to replicate it I got the following error:

> The error occurred in attempting to run the following R command(s):
> rm(.temp_value)

2. I can write in the "Format Description" text box which is probably not ideal

**Comment by lilyclements** [comment]
Fixed the second issue (waiting to merge with other changes)

Haven't fixed the first

**Comment by lilyclements** [comment]
@stevenndungu noticed a quick small thing to fix. I fixed it here:
https://github.com/stevenndungu/R-Instat/pull/299#pullrequestreview-48782488

If you've already started to make changes to this branch then don't worry about merging in mine. Just add the line I've changed into yours to fix the problem

**Comment by stevenndungu** [comment]
@lilyclements "I clicked "Help..." and first time it crashed. When I tried to replicate it I got the following error:"
I am not able to get this error. I have tried severally. Not sure what could have happened. The code looks similar to the help button in Import dialog which does not bug.

Could you update your branch and see if you can reproduce the error again. 
Thanks for making the txtPreview read-only...that was important

**Comment by lilyclements** [comment]
I can't replicate it anymore. I'll try hard and crash it tomorrow but it should be all good now for Danny to review

**Pull request merged**